Makefile

```diff
@@ -177,8 +177,8 @@ CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
 ENVTEST ?= $(LOCALBIN)/setup-envtest
 
 ## Tool Versions
-KUSTOMIZE_VERSION ?= v5.1.1
-CONTROLLER_TOOLS_VERSION ?= v0.16.1
+KUSTOMIZE_VERSION ?= v5.7.1
+CONTROLLER_TOOLS_VERSION ?= v0.19.0
```

Comment:
`v0.19.0` allows using the `+kubebuilder:validation:ExactlyOneOf` marker. Haven't tried it, but sounds useful to simplify https://github.com/ionos-cloud/cluster-api-provider-ionoscloud/blob/e2234880ab5219ffdb60ce9f8eb192db316675fe/api/v1alpha1/ionoscloudmachine_types.go#L237-L238